OpenWeather/App/Search/SearchScreen.swift

```diff
@@ -17,71 +17,54 @@
 import OpenBytesNavigation
 import SwiftUI
 
-struct SearchScreen: View {
+struct LocationResultItem: View {
+    let name: String
+
     var body: some View {
-        List {
-            Button(
-                action: { Navigation.path.push(Date()) },
-                label: { Text("Push") }
-            )
+        VStack {
+            Text(name)
 
-            Button(
-                action: { Navigation.path.modal { Text("Modal") } },
-                label: { Text("Modal") }
-            )
+            // ...
+        }
+    }
+}
 
-            Button(
-                action: {
-                    Navigation.path.alert(
-                        title: Text("Alert Title"),
-                        primaryButton: .default(Text("Done")),
-                        secondaryButton: .cancel()
-                    )
-                },
-                label: { Text("Alert") }
-            )
+struct SearchScreen: View {
+    @State private var searchText: String = ""
+    @ObservedObject var viewModel: SearchViewModel
 
-            Button(
-                action: {
-                    Navigation.path.actionSheet(
-                        title: "Action Sheet Title",
-                        actions: {
-                            Button(
-                                action: { print("Hi") },
-                                label: { Text("Say Hi") }
-                            )
-                        },
-                        message: { Text("Action Sheet Message") }
-                    )
-                },
-                label: { Text("Action Sheet") }
-            )
+    init(viewModel: SearchViewModel) {
+        self.viewModel = viewModel
+    }
 
-            Button {
-                Navigation.path.toast(
-                    title: "Toast Title",
-                    message: "This is a toast message and can be fairly long if needed.",
-                    style: .error
-                )
-            } label: {
-                Text("Toast Alert")
+    var body: some View {
+        List {
+            ForEach(viewModel.result) { location in
+                LocationResultItem(name: location.name)
             }
-
         }
+        .searchable(text: $searchText, prompt: "Search by name or zipcode")
         .navigationDestination(
             for: Date.self,
             destination: { date in
                 Text(date, format: .dateTime)
             }
         )
         .navigationTitle("Search")
+        .onChange(of: searchText) { newSearchText in
+            viewModel.getLocations(query: newSearchText)
+        }
```

Comment:
Ideally this should live in the ViewModel.. 

Also we might want to add some debouncing.. 

Since if the user types in "12345" we would cancel and fire new tasks for each character typed.. If we add some debouncing we could delay by 0.3 seconds and then we would only fire once.

Reply:
Ran into this issue with an app, and found using a submit button was the fix because even adding a delay caused my app to lock up. It may have been my implementation.

Reply:
@0xLeif Oh I see, I know about this issue, I used to solve it using custom `Publisher` on `UITextField`, will see how I accomplish this in SwiftUI.